queens/iterators/control_variates_iterator.py

```diff
@@ -0,0 +1,246 @@
+        seed (int): Seed for random samples.
+        num_samples (int):  Number of samples on the main estimator
+        expectation_cvs (float):    Expectation of the control variate. If the expectation is None,
+                                    it will be estimated via MC sampling.
+        output (dict):  Output dict with the following entries:
+                        * ``mean`` (float): Estimated mean of main model.
+                        * ``std`` (float): Estimated standard deviation of the main model mean
+                        estimate.
+                        * ``num_samples_cv`` (int): Number of samples on control variate.
+                        * ``mean_cv`` (float): Mean of control variate.
+                        * ``std_cv`` (float): Standard deviation of control variate.
+                        * ``alpha`` (float): Method specific parameter that determines the
+                        influence of control variate.
+                        * ``sample_ratio`` (float): Ratio of number of samples on control variate to
+                                            number of samples on main model. Is only part of output
+                                            if use_optimal_num_samples==True.
+        num_samples_cv (int):    Number of samples to use for computing the expectation
+                                        value of control variate. If expectation of the control
+                                        variate is known, you can pass None as it's number of
+                                        samples.
+        samples (np.array): Samples for which all models are evaluated on.
+        use_optimal_num_samples (bool):    Determines if the iterator calculates and uses the
+                            optimal number of samples.
+        cost_model (float): Cost of evaluating the model.
+        cost_cv (float):    Cost of evaluating the control variate.
+        variance_cv_mean_estimator (float):   variance of the control variate mean estimator
+    """
+
+    @log_init_args
+    def __init__(
+        self,
+        model,
+        control_variate,
+        parameters,
+        global_settings,
+        seed,
+        num_samples,
+        expectation_cv=None,
+        num_samples_cv=None,
+        use_optimal_num_samples=False,
+        cost_model=None,
+        cost_cv=None,
+    ):
+        """Initialize the control variates iterator.
+
+        Args:
+            model (Model):  The main model. Uncertainties are quantified for this model.
+            control_variate (Model):    The control variate - a model that approximates the main
+            model and is cheaper to evaluate.
+            parameters (Parameters):    Parameters to use for evaluation.
+            global_settings (GlobalSettings):   Global settings to use for evaluation.
+            seed (int): Seed for random samples.
+            num_samples (int):  Number of samples to evaluate for estimation.
```

Comment:
```suggestion
            model (Model): The main model. Uncertainties are quantified for this model.
            control_variate (Model): The control variate - a model that approximates the main
                                     model and is cheaper to evaluate.
            parameters (Parameters):  Parameters to use for evaluation.
            global_settings (GlobalSettings): Global settings to use for evaluation.
            seed (int): Seed for random samples.
            num_samples (int): Number of samples to evaluate for estimation.
```
Please check that your indentations are correct in the future. Also, update the argument descriptions here according the the suggestions I provided for the attributes.

Reply:
I’ll be more careful about indentations in the future. To avoid any confusion, what is the correct method of indentation here? I am assuming one whitespace between the colon and the description of the argument? During coding, I used the Monte Carlo iterator as a reference for styling. I don’t think this indentation convention is followed in the Monte Carlo Iterator. Which iterator is on your gold standard for styling that I can use for future reference?

Reply:
Yes, one whitespace between the colon and the description, and if the description is longer than one line, the new lines always start wherever the first line started (irrespective of where the other descriptions start).

That is a fair point; we haven't always applied this rule in the past. At first glance, the MetropolisHastingsIterator seems to follow the rule I described above.